Added support for shard allocation awareness by zone and forced awareness in Helm chart

juntezhang opened the pull request:
*Issue:*

https://github.com/opendistro-for-elasticsearch/opendistro-build/issues/728

*Description of changes:*

Added support for Shard Allocation Awareness by Zone and Forced Allocation Awareness of the shards by zones.

*Test Results:*

Manual testing done by checking for the added configuration properties in the `_nodes` endpoint. 

By enabling `elasticsearch.enableShardAllocationAwareness`, we see that the following properties have been added:
- `node.attr.zone`: <ZONE>. If there is no zone configuration, then this value will be `0`.
- `cluster.routing.allocation.awareness.attributes: zone`

By enabling `elasticsearch.forcedAllocationAwareness`, we see that the following propery has also been added:
- `cluster.routing.allocation.awareness.force.zone.values: <ZONES>`, where `<ZONES>` is a CSV of all zones configured within the node pools. 

By disabling `elasticsearch.enableShardAllocationAwareness`, we don't see any of the above properties being added.

By disabling `elasticsearch.enableShardAllocationAwareness` and only enabling `elasticsearch.forcedAllocationAwareness`, we don't see any of the above properties being added.

peterzhuamazon commented:
Hi @juntezhang thanks for creating this PR.

After reviewing the code, we have found this [docker image](https://hub.docker.com/r/bskim45/helm-kubectl-jq) linked in the code has no descriptions and certification on docker hub, and raise concern on the legitimacy of it.

Could you find an alternative solution to this?

Thanks.

juntezhang commented:
> After reviewing the code, we have found this [docker image](https://hub.docker.com/r/bskim45/helm-kubectl-jq) linked in the code has no descriptions on docker hub, and raise concern on the legitimacy/safety of it.
> 
> Could you find an alternative solution to this?

I have created another Docker image here and added the commands used to build : https://hub.docker.com/r/juntezhang/kubectl-jq

I have added the commands used to build this image. And I tested it, still works. Perhaps you could host this image instead, if you prefer?

peterzhuamazon commented:
> > After reviewing the code, we have found this [docker image](https://hub.docker.com/r/bskim45/helm-kubectl-jq) linked in the code has no descriptions on docker hub, and raise concern on the legitimacy/safety of it.
> > Could you find an alternative solution to this?
> 
> I have created another Docker image here and added the commands used to build : https://hub.docker.com/r/juntezhang/kubectl-jq
> 
> And I tested it, still works. Perhaps you could host this image instead, if you prefer?

Hi @juntezhang is there any official images instead of images form individual account?
The reason being this Helm Chart can be used in Production Environment where individual docker image is discouraged.

Thanks.

juntezhang commented:
Thanks for the update @sreekarjami . I forgot about this PR to be honest. We could use a generic image and then install the tools there, but for us, we prefer to have a pre-made image. I have added the commands in my docker image on how to create this image. 

This PR can be a reference for others who want to have shard allocation awareness and forced awareness supported in this Helm chart. I am also open to discover better solutions, but so far, I have not found a better solution yet.
